freqtrade/optimize/__init__.py

```diff
@@ -4,25 +4,46 @@
 import json
 import logging
 import os
+import arrow
 from typing import Optional, List, Dict, Tuple
 
-from freqtrade import misc
+from freqtrade import misc, constants
 from freqtrade.exchange import get_ticker_history
+
 from user_data.hyperopt_conf import hyperopt_optimize_conf
 
 logger = logging.getLogger(__name__)
 
 
 def trim_tickerlist(tickerlist: List[Dict], timerange: Tuple[Tuple, int, int]) -> List[Dict]:
+    if not tickerlist:
+        return tickerlist
```

Comment:
What is the intent here? 
if `None` is given, return `None`?

Reply:
If tickerlist is an empty list "[]" return it. It could happen if there is no data for the requested pair.

Reply:
Shouldn't it then be better to return `[]` if for some reason `tickerlist` will be `None`?

Reply:
I think it's fine to just return the parameter if empty. tickerlist should never be `None` based on the typehint, if so the invocation should fixed or the typehint adapted.